plugins/field-bitmap/test/field_test.mocha.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const {testHelpers} = require('@blockly/dev-tools');
+const {assert} = require('chai');
+const {FieldBitmap, DEFAULT_HEIGHT, DEFAULT_WIDTH} = require('../src/index');
+
+/**
+ * Helper method to reformat raw test cases to a format that the
+ * various tests can use.
+ * @param {*} testCases
+ * @return the processed test cases
+ */
+function processTestCases(testCases) {
+  return testCases.map((x) => ({
+    title: x.title,
+    value: x.value,
+    args: [x.value, null, {}],
+    json: {value: x.value},
+  }));
+}
+
+const {
+  FieldCreationTestCase,
+  FieldValueTestCase,
+  runConstructorSuiteTests,
+  runFromJsonSuiteTests,
+} = testHelpers;
+
+suite('FieldBitmap', function() {
+  /**
+   * Configuration for field tests with invalid values.
+   * @type {Array<FieldCreationTestCase>}
+   */
+  const invalidValueTestCases = processTestCases([
+    {title: 'Undefined', value: undefined},
+    {title: 'Null', value: null},
+    {title: 'NaN', value: NaN},
+    {title: 'Non-Parsable String', value: 'bad'},
+    {title: 'Not an array', value: 1},
+    {title: 'Not a 2D array', value: [0, 1, 0, 1]},
+    {
+      title: 'Not a rectangle',
+      value: [[1, 1, 1], [1, 1, 1, 1, 1, 1], [1, 1, 1]],
+    },
+    {
+      title: 'Contains non-binary number',
+      value: [[1, 99, 1], [1, 1, 1], [1, 1, 1]],
+    },
+    {
+      title: 'Contains bad value',
+      value: [[1, 'b', 1], [1, 1, 1], [1, 1, 1]],
+    },
+  ]);
+
+  /**
+   * Configuration for field tests with valid values.
+   * @type {Array<FieldCreationTestCase>}
+   */
+  const validValueTestCases = processTestCases([
+    {title: '3x3 solid', value: [[1, 1, 1], [1, 1, 1], [1, 1, 1]]},
+    {
+      title: '4x4 solid',
+      value: [[1, 1, 1, 1], [1, 1, 1, 1], [1, 1, 1, 1], [1, 1, 1, 1]],
+    },
+    {title: '3x3 checkerboard', value: [[1, 0, 1], [0, 1, 0], [1, 0, 1]]},
+    {
+      title: '4x4 solid',
+      value: [[0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]],
+    },
+  ]);
+
+  /**
+   * The expected default value for the field being tested.
+   * @type {*}
+   */
+  const defaultFieldValue = [
+    [0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+  ];
+  /**
+   * Asserts that the field property values are set to default.
+   * @param {FieldBitmap} field The field to check.
+   */
+  const assertFieldDefault = function(field) {
+    assert(
+        JSON.stringify(field.getValue()) === JSON.stringify(defaultFieldValue));
```

Comment:
I think you can use [assert.deepEqual](http://aaronsofaly.github.io/chai-docs/api/assert/#method_deepequal) here.